python/semantic_kernel/agents/azure_ai/azure_ai_agent.py

```diff
@@ -156,7 +341,8 @@ async def get_messages(self, sort_order: Literal["asc", "desc"] = "desc") -> Asy
 
 
 @experimental
-class AzureAIAgent(Agent):
+@register_agent_type("foundry_agent")
+class AzureAIAgent(DeclarativeSpecMixin, Agent):
```

Comment:
```suggestion
class AzureAIAgent(Agent, DeclarativeSpecMixin):
```
Not sure if it matters, but for me the base class should always be first.

Reply:
We need need DeclarativeSpecMixin first so its from_dict and resolve_placeholders methods are prioritized in the MRO. Flipping the order would prioritize the Agent base class in the MRO. But Agent doesn't implement from_dict, from_yaml, or resolve_placeholders -- those live in DeclarativeSpecMixin.